inference-engine/tests/functional/plugin/myriad/ngraph/transformations/dynamic_to_static_shape_non_max_suppression_opset3.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <common_test_utils/test_common.hpp>
+#include <ngraph_functions/utils/ngraph_helpers.hpp>
+#include <ngraph/opsets/opset3.hpp>
+#include <vpu/ngraph/operations/dynamic_shape_resolver.hpp>
+#include <vpu/ngraph/transformations/dynamic_to_static_shape.hpp>
+#include <vpu/ngraph/transformations/dynamic_to_static_shape_non_max_suppression.hpp>
+#include <vpu/utils/error.hpp>
+#include <numeric>
+#include <queue>
+#include <random>
+
+namespace {
+
+using DataType = ngraph::element::Type_t;
+using DataDims = ngraph::Shape;
+
+struct NonMaxSuppressionTestCase {
+    int64_t num_batches, num_boxes, num_classes, max_output_boxes_per_class;
+    float iou_threshold, score_threshold;
+};
+
+
+class DynamicToStaticShapeNonMaxSuppressionOpset3 : public CommonTestUtils::TestsCommon,
+        public testing::WithParamInterface<std::tuple<DataType, DataType, NonMaxSuppressionTestCase>> {
+public:
+    void SetUp() override {
+        const auto& parameters = GetParam();
+        const auto& float_type = std::get<0>(parameters);
+        const auto& integer_type = std::get<1>(parameters);
+        const auto& nms_setup = std::get<2>(parameters);
+
+        ngraph::helpers::CompareFunctions(*transform(float_type, integer_type, nms_setup),
+                *reference(float_type, integer_type, nms_setup));
+    }
+
+protected:
+    std::shared_ptr<const ngraph::Function> transform(
+            const ngraph::element::Type_t& float_type,
+            const ngraph::element::Type_t& integer_type,
+            const NonMaxSuppressionTestCase& nms_setup) const {
+        const auto boxes = std::make_shared<ngraph::opset3::Parameter>(
+                float_type, ngraph::PartialShape{nms_setup.num_batches, nms_setup.num_boxes, 4});
+        const auto scores = std::make_shared<ngraph::opset3::Parameter>(
+                float_type, ngraph::PartialShape{nms_setup.num_batches, nms_setup.num_classes, nms_setup.num_boxes});
+        const auto max_output_boxes_per_class = ngraph::opset3::Constant::create(integer_type, {}, std::vector<int64_t>{nms_setup.max_output_boxes_per_class});
+        const auto iou_threshold = ngraph::opset3::Constant::create(float_type, ngraph::Shape{}, std::vector<float>{nms_setup.iou_threshold});
+        const auto score_threshold = ngraph::opset3::Constant::create(float_type, ngraph::Shape{}, std::vector<float>{nms_setup.score_threshold});
```

Comment:
Is there common logic in generating parameters/constants for `transform` and `reference`? Is it reasonable to extract it in a separate method?
Does test case have something in common with those we see in Faster/Mask-RCNN? I think it's a good idea to place here subgraph/parameters for operations from targeted models..